internal/pkg/service/buffer/api/service/service.go

```diff
@@ -168,7 +168,42 @@ func (s *service) CreateExport(d dependencies.ForProjectRequest, payload *buffer
 }
 
 func (s *service) UpdateExport(d dependencies.ForProjectRequest, payload *buffer.UpdateExportPayload) (res *buffer.Export, err error) {
-	return nil, NewNotImplementedError()
+	ctx, str, mpr := d.RequestCtx(), d.Store(), s.mapper
+
+	// Get export
+	exportKey := key.ExportKey{
+		ReceiverKey: key.ReceiverKey{
+			ProjectID:  d.ProjectID(),
+			ReceiverID: string(payload.ReceiverID),
+		},
+		ExportID: string(payload.ExportID),
+	}
+
+	old, err := str.GetExport(ctx, exportKey)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update
+	export, err := mpr.UpdateExportFromPayload(old, *payload)
+	if err != nil {
+		return nil, err
+	}
+
+	// Persist
+	err = str.UpdateExport(ctx, export)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Mno, tohle je možná taky skoro na transakci. 🤔 I když asi ne tak nutně jako ta předchozí situace. Tady by došlo maximálně tak k race conditions mezi updatama s čímž se asi dá žít. (Kdežto ten předchozí delete a insert by mohl způsobit že když se dobře trefíš, tak dostaneš na GET 404ku.)

Reply:
urcite cez transakciu, podobne ako tu https://github.com/keboola/keboola-as-code/pull/1037#discussion_r1036095657 (ak by si sa tam nejak zasekol, tak daj vediet) cc @jprochazk

Reply:
Medzi create a update nie je prakticky ziadny rozdiel, len ze pri create pouzivame `PutIfNotExists`, a pri update `Put`, ideal to spravit na nejaku option ako som pisal vyssie.

Reply:
Ak transakcia spadne, tak treba vratit nejaky rozumny HTTP kod s hlaskou, ze sa dana hodnota zmenila, a je potrebne request zopakovat.

Reply:
Jaký je v tom benefit? Máme spousty těch ops co dělají jen `schema.Configs().Something()` a rozdíl mezi nimi je jen na konci `Put`/`Delete`/atd. Proč se to musí zbytečně víc abstrahovat?

Reply:
Uplne PRO level by bol, ak by sme aj v GET response vracali `revision`, a v `PATCH` response umoznovali zadat `revision` HEADER, ... tak by to bolo E2E atomicke, ... aj ak by klient menil hodnotu pol hodiny.

Reply:
> Jaký je v tom benefit? 

Na co konretne sa pytas, na tu `revision`?

Mas 2 requesty, jeden meni `name`, druhy `conditions`, nezavysle od seba.

Poslu sa naraz, ... obe si nacitaju revision `X`.

Request co meni `name` spravi `PUT` a prepise CELY kluc na novu revision `Y`. HTTP request dobehne s tym , ze sa hodnota zmenila.

Druhy request tiez spravi `PUT` ale o cosi neskor, ... zmeni revision na `Z`, ... sice nastavi spravne/nove `conditions`, ale nastavi tam stare `name` a klient napise na ZD ....

Spravne ma ten 2 request: spadnut, alebo spravit retry, ale sucastou retry musi byt aj ten GET.

Dava to zmysel? @jprochazk

Reply:
Teda je to funkcna poziadavka, nie len nejake vylepsenie kodu.
Ako to dosiahnes je na tebe, mozno najdes lepsi sposob.

Reply:
> Medzi create a update nie je prakticky ziadny rozdiel, len ze pri create pouzivame PutIfNotExists, a pri update Put, ideal to spravit na nejaku option ako som pisal vyssie.

Reply:
> Medzi create a update nie je prakticky ziadny rozdiel, len ze pri create pouzivame PutIfNotExists, a pri update Put, ideal to spravit na nejaku option ako som pisal vyssie.

Kludne to nakopiruj, to je jedno, to nebola hlavna pointa, iba navrh ako to mozes spravit.

Reply:
> Teda je to funkcna poziadavka, nie len nejake vylepsenie kodu.
Ako to dosiahnes je na tebe, mozno najdes lepsi sposob.

Dává to smysl, ale je to docela podstatný rozdíl od toho, jak to zrovna teď funguje. Co se týče toho, jestli je to API E2E atomické, tak určitě není a ta revision není jediný problém. To ale prosím řešme v klidu v jiném PR. Tady jde hlavně o to, aby bylo API funkční E2E.

Reply:
ok, kludne v dalsom, toto sa moze mergnut

dolezite je si este uvedomit, ze sa to neda vyriesit iba zamkom, ale prave tou `revision`.
https://etcd.io/docs/v3.5/learning/why/#notes-on-the-usage-of-lock-and-lease

```
Then how does the lease mechanism guarantees mutual exclusion of the locking mechanism? Actually, the lease mechanism itself doesn’t guarantee mutual exclusion. Owning a lease cannot guarantee the owner holds a lock of the resource.
```

Reply:
> API E2E atomické, tak určitě není a ta revision není jediný problém.

No niektore endpointy (create) su atomicke, treba doriesit este ostatne.